sdk/core/System.ClientModel/src/Message/BinaryContent.cs

```diff
@@ -219,9 +219,16 @@ public StreamBinaryContent(Stream stream)
 
         public override bool TryComputeLength(out long length)
         {
-            // CanSeek is validated in constructor - it will always be true.
-            length = _stream.Length - _origin;
-            return true;
+            // CanSeek was checked in the type constructor, but it can
+            // change based on the state of the stream.
+            if (_stream.CanSeek)
+            {
+                length = _stream.Length - _origin;
```

Comment:
Can you remind me why we don't provide the length directly as opposed to giving the length remaining after the current stream position?

Reply:
I'm not 100% confident of the original author's intent, but I had assumed it was in case someone was writing to a stream starting from some non-zero position.  I've been poking around the BCL HttpContent code, and it looks like things like MultipartContent compute their Content-Length by calling this method on their "parts" and then adding in the length of the content segments that they add when serializing to a stream.  So, it could be possible that the full stream length (_stream.Length) represents more than the length of this instance's length.